backend/graphql/types/courseType.ts

```diff
@@ -0,0 +1,33 @@
+import { gql } from "apollo-server-express";
+
+const courseType = gql`
+  type CourseResponseDTO {
+    id: ID!
+    title: String!
+    description: String
+    image: String
+    previewImage: String
+    lessons: [ID!]!
```

Comment:
Just so I understand, does this mean there has to be a lesson ID within the list? Can the list be be empty?

Reply:
From the docs (https://graphql.org/learn/schema/):
> [Episode!]! represents an array of Episode objects. Since it is also non-nullable, you can always expect an array (with zero or more items) when you query the appearsIn field. And since Episode! is also non-nullable, you can always expect every item of the array to be an Episode object.

Sounds like it guarantees it has to be an array always, and content of the array has to be that type always

Reply:
Whoops, I got confused @Shrinjay. I thought this was the lesson API PR. `lessons` will indeed be a list of IDs. Sorry for the confusion!